src/Microsoft.DotNet.XHarness.Tests.Runners/AndroidApplicationEntryPoint.cs

```diff
@@ -0,0 +1,78 @@
+﻿
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.Tests.Runners.Core;
+
+namespace Microsoft.DotNet.XHarness.Tests.Runners
+{
+    public abstract class AndroidApplicationEntryPoint : ApplicationEntryPoint
+    {
+        /// <summary>
+        /// Implementators should provide a text writter than will be used to
+        /// write the logging of the test that are executed.
+        /// </summary>
+        public abstract TextWriter Logger { get; }
+
+        /// <summary>
+        /// Implementators should provide a full path in which the final 
+        /// results of the test run will be written. This property must not
+        /// return null.
+        /// </summary>
+        public abstract string TestsResultsFinalPath { get; }
+
+
+        public override async Task RunAsync()
+        {
+            var options = ApplicationOptions.Current;
+            // we generate the logs in two different ways depending if the generate xml flag was
+            // provided. If it was, we will write the xml file to the tcp writer if present, else
+            // we will write the normal console output using the LogWriter
+            var logger = (Logger == null || options.EnableXml) ? new LogWriter(Device) : new LogWriter(Device, Logger);
+            logger.MinimumLogLevel = MinimumLogLevel.Info;
+            var testAssemblies = GetTestAssemblies();
+
+            var runner = await CreateRunner(logger);
+
+            // if we have ignore files, ignore those tests
+            await runner.Run(testAssemblies).ConfigureAwait(false);
+
+            TestRunner.Jargon jargon = Core.TestRunner.Jargon.NUnitV3;
```

Comment:
we definitely don't want to use NUnit xml on Helix but xunit

Reply:
@akoeplinger you can pass the jargon to use in the app arguments. One question, does the helix test uploaded take the attachments in xunit xml? That might mean I have to add some code for the xml generation, not hard but we should check it

Reply:
Helix is fine with XUnit or NUnit to report to Azure Devops.  For Kusto tracking (legacy) it's got to be XUnit.